application/migrations/010_add_long_running_flag.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+defined('BASEPATH') OR exit('No direct script access allowed');
+
+class Migration_add_long_running_flag extends CI_Migration
+{
+    // Add a column to indicate whether the office's crawl takes longer than 23 hours
+    public function up()
+    {
+        // add our column if it doesn't exist
+        $this->CI->db->query("CALL add_column_if_not_exists('offices', 'long_running', 'varchar(256)')");
+        // test if it exists
+        $query = $this->CI->db->query("SELECT column_exists('offices', 'long_running')");
+        $row = $query->first_row('array');
+        $tableAndColumnExist = (bool) array_shift($row);
+        // add values if it exists
+        if ($tableAndColumnExist) {
+            // set out default
+            $this->db->query("UPDATE offices SET long_running = 'FALSE' ");
```

Comment:
This should be done by setting a default on the added column above. Otherwise it's possible someone might add a new row and manage to get a NULL value in there.